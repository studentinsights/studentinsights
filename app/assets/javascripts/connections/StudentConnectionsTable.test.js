import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import _ from 'lodash';
import * as Virtualized from 'react-virtualized';
import {withDefaultNowContext} from '../testing/NowContainer';
import unvirtualize from '../testing/unvirtualize';
import StudentConnectionsTable, {orderedStudents} from './StudentConnectionsTable';
import connectionsShowJson from './connectionsShowJson.fixture';
import PerDistrictContainer from '../components/PerDistrictContainer';
const {SortDirection} = Virtualized;


// can't add snapshots with react-virtualized without
// some work, see https://github.com/bvaughn/react-virtualized/issues/1117
// instead, we stub these implementations with naive simple unvirtualized
// ones, to verify the substance of the rendering works
jest.mock('react-virtualized');
unvirtualize(Virtualized, jest.fn);


function testProps(props = {}) {
  return {
    orderedStudentsWith2020Surveys: connectionsShowJson.students_with_2020_survey_data,
    sortBy: 'grade',
    sortDirection: SortDirection.ASC,
    onTableSort: jest.fn(),
    ...props
  };
}

it('renders without crashing', () => {
  const props = testProps();
  const el = document.createElement('div');
  ReactDOM.render(withDefaultNowContext(
    <PerDistrictContainer districtKey="somerville">
      <StudentConnectionsTable {...props} />
    </PerDistrictContainer>
  ), el);
  expect($(el).find('.StudentConnectionsTable').length).toEqual(1);
});

describe('orderedStudents', () => {
  it('works', () => {
    const filteredStudents = connectionsShowJson.students_with_2020_survey_data;
    const orderedStudentsList = orderedStudents(filteredStudents, 'science', SortDirection.ASC);
    const orderedGrades = _.compact(_.flatten(orderedStudentsList.map(student => {
      return student.student_section_assignments_right_now.filter(assignment => {
        return assignment.section.id == 4;
      }).map(assignment => assignment.grade_letter);
    })));
    expect(orderedGrades).toEqual(['D', 'D', 'C', 'D']);
  });
});


it('snapshots with unvirtualized render', () => {
  const props = testProps();
  const tree = renderer
    .create(withDefaultNowContext(
      <PerDistrictContainer districtKey="somerville">
      <StudentConnectionsTable {...props} />
      </PerDistrictContainer>))
    .toJSON();
  expect(tree).toMatchSnapshot();
});
