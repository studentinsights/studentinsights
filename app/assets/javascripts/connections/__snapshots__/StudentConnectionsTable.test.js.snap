// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots with unvirtualized render 1`] = `
<div
  className="StudentConnectionsTable"
  style={
    {
      "display": "block",
      "flex": 1,
      "flexDirection": "column",
      "overflow": "hidden",
    }
  }
>
  <div
    className="Mock-AutoSizer"
  >
    <table
      className="MockReactVirtualized"
    >
      <thead>
        <tr>
          <td>
            Student
          </td>
          <td>
            Is there an adult at SHS you can reach out to if there are any issues?
          </td>
          <td>
            Absences
          </td>
          <td>
            Discipline Incidents
          </td>
          <td>
            EN or ELL
          </td>
          <td>
            Social Studies
          </td>
          <td>
            Math
          </td>
          <td>
            Science
          </td>
          <td>
            Last xGE or NEST
          </td>
          <td>
            Last SST
          </td>
          <td>
            Last Counselor
          </td>
          <td>
            Study skills
          </td>
          <td>
            Academic Support
          </td>
          <td>
            Redirect
          </td>
          <td>
            Credit Recovery
          </td>
          <td>
            Program or SPED
          </td>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            <div
              style={
                {
                  "alignItems": "center",
                  "display": "flex",
                  "justifyContent": "flex-start",
                }
              }
            >
              <div
                className="StudentPhotoCropped"
                style={
                  {
                    "backgroundColor": "#ddd",
                    "backgroundImage": "url(/students/22/photo)",
                    "backgroundPosition": "center top",
                    "backgroundPositionY": -10,
                    "backgroundRepeat": "no-repeat",
                    "backgroundSize": "150%",
                    "border": "1px solid #ddd",
                    "borderRadius": 3,
                    "height": 58,
                    "marginRight": 10,
                    "overflow": "hidden",
                    "width": 58,
                  }
                }
              />
              <a
                href="/students/22"
                rel="noopener noreferrer"
                style={
                  {
                    "fontSize": 14,
                    "fontWeight": "bold",
                  }
                }
                target="_blank"
              >
                Snow
                 
                Kenobi
              </a>
            </div>
          </td>
          <td>
            <span
              style={
                {
                  "textAlign": "center",
                }
              }
            >
              no
            </span>
          </td>
          <td>
            <span
              style={
                {
                  "display": "inline-block",
                  "padding": 8,
                }
              }
            >
              0
            </span>
          </td>
          <td>
            <span
              style={
                {
                  "display": "inline-block",
                  "padding": 8,
                }
              }
            >
              0
            </span>
          </td>
          <td />
          <td />
          <td />
          <td />
          <td />
          <td />
          <td />
          <td />
          <td />
          <td />
          <td />
          <td>
            <span
              style={
                {
                  "backgroundColor": "rgba(51, 51, 51, 0.5)",
                  "color": "white",
                  "display": "inline-block",
                  "fontSize": 12,
                  "height": "95%",
                  "padding": 4,
                  "textAlign": "center",
                  "width": "95%",
                }
              }
            >
              2Way Spanish
            </span>
          </td>
        </tr>
        <tr>
          <td>
            <div
              style={
                {
                  "alignItems": "center",
                  "display": "flex",
                  "justifyContent": "flex-start",
                }
              }
            >
              <div
                className="StudentPhotoCropped"
                style={
                  {
                    "backgroundColor": "#ddd",
                    "backgroundImage": "url(/students/25/photo)",
                    "backgroundPosition": "center top",
                    "backgroundPositionY": -10,
                    "backgroundRepeat": "no-repeat",
                    "backgroundSize": "150%",
                    "border": "1px solid #ddd",
                    "borderRadius": 3,
                    "height": 58,
                    "marginRight": 10,
                    "overflow": "hidden",
                    "width": 58,
                  }
                }
              />
              <a
                href="/students/25"
                rel="noopener noreferrer"
                style={
                  {
                    "fontSize": 14,
                    "fontWeight": "bold",
                  }
                }
                target="_blank"
              >
                Winnie
                 
                Kenobi
              </a>
            </div>
          </td>
          <td>
            <span
              style={
                {
                  "textAlign": "center",
                }
              }
            >
              no,
            </span>
          </td>
          <td>
            <span
              style={
                {
                  "display": "inline-block",
                  "padding": 8,
                }
              }
            >
              0
            </span>
          </td>
          <td>
            <span
              style={
                {
                  "display": "inline-block",
                  "padding": 8,
                }
              }
            >
              0
            </span>
          </td>
          <td />
          <td />
          <td />
          <td />
          <td />
          <td />
          <td />
          <td />
          <td />
          <td />
          <td />
          <td>
            <span
              style={
                {
                  "backgroundColor": "rgba(51, 51, 51, 0.5)",
                  "color": "white",
                  "display": "inline-block",
                  "fontSize": 12,
                  "height": "95%",
                  "padding": 4,
                  "textAlign": "center",
                  "width": "95%",
                }
              }
            >
              2Way English
            </span>
          </td>
        </tr>
        <tr>
          <td>
            <div
              style={
                {
                  "alignItems": "center",
                  "display": "flex",
                  "justifyContent": "flex-start",
                }
              }
            >
              <div
                className="StudentPhotoCropped"
                style={
                  {
                    "backgroundColor": "#ddd",
                    "backgroundImage": "url(/students/26/photo)",
                    "backgroundPosition": "center top",
                    "backgroundPositionY": -10,
                    "backgroundRepeat": "no-repeat",
                    "backgroundSize": "150%",
                    "border": "1px solid #ddd",
                    "borderRadius": 3,
                    "height": 58,
                    "marginRight": 10,
                    "overflow": "hidden",
                    "width": 58,
                  }
                }
              />
              <a
                href="/students/26"
                rel="noopener noreferrer"
                style={
                  {
                    "fontSize": 14,
                    "fontWeight": "bold",
                  }
                }
                target="_blank"
              >
                Pluto
                 
                White
              </a>
            </div>
          </td>
          <td>
            <span
              style={
                {
                  "textAlign": "center",
                }
              }
            >
              no, I don't feel comfortable with anyone
            </span>
          </td>
          <td>
            <span
              style={
                {
                  "display": "inline-block",
                  "padding": 8,
                }
              }
            >
              0
            </span>
          </td>
          <td>
            <span
              style={
                {
                  "display": "inline-block",
                  "padding": 8,
                }
              }
            >
              0
            </span>
          </td>
          <td />
          <td />
          <td />
          <td />
          <td />
          <td />
          <td />
          <td />
          <td />
          <td />
          <td />
          <td />
        </tr>
        <tr>
          <td>
            <div
              style={
                {
                  "alignItems": "center",
                  "display": "flex",
                  "justifyContent": "flex-start",
                }
              }
            >
              <div
                className="StudentPhotoCropped"
                style={
                  {
                    "backgroundColor": "#ddd",
                    "backgroundImage": "url(/students/28/photo)",
                    "backgroundPosition": "center top",
                    "backgroundPositionY": -10,
                    "backgroundRepeat": "no-repeat",
                    "backgroundSize": "150%",
                    "border": "1px solid #ddd",
                    "borderRadius": 3,
                    "height": 58,
                    "marginRight": 10,
                    "overflow": "hidden",
                    "width": 58,
                  }
                }
              />
              <a
                href="/students/28"
                rel="noopener noreferrer"
                style={
                  {
                    "fontSize": 14,
                    "fontWeight": "bold",
                  }
                }
                target="_blank"
              >
                Rapunzel
                 
                Mouse
              </a>
            </div>
          </td>
          <td>
            <span
              style={
                {
                  "textAlign": "center",
                }
              }
            >
              Yes. Mrs. Teacher
            </span>
          </td>
          <td>
            <span
              style={
                {
                  "display": "inline-block",
                  "padding": 8,
                }
              }
            >
              0
            </span>
          </td>
          <td>
            <span
              style={
                {
                  "display": "inline-block",
                  "padding": 8,
                }
              }
            >
              0
            </span>
          </td>
          <td />
          <td />
          <td />
          <td />
          <td />
          <td />
          <td />
          <td />
          <td />
          <td />
          <td />
          <td>
            <span
              style={
                {
                  "backgroundColor": "rgba(51, 51, 51, 0.5)",
                  "color": "white",
                  "display": "inline-block",
                  "fontSize": 12,
                  "height": "95%",
                  "padding": 4,
                  "textAlign": "center",
                  "width": "95%",
                }
              }
            >
              2Way Spanish
            </span>
          </td>
        </tr>
        <tr>
          <td>
            <div
              style={
                {
                  "alignItems": "center",
                  "display": "flex",
                  "justifyContent": "flex-start",
                }
              }
            >
              <div
                className="StudentPhotoCropped"
                style={
                  {
                    "backgroundColor": "#ddd",
                    "backgroundImage": "url(/students/31/photo)",
                    "backgroundPosition": "center top",
                    "backgroundPositionY": -10,
                    "backgroundRepeat": "no-repeat",
                    "backgroundSize": "150%",
                    "border": "1px solid #ddd",
                    "borderRadius": 3,
                    "height": 58,
                    "marginRight": 10,
                    "overflow": "hidden",
                    "width": 58,
                  }
                }
              />
              <a
                href="/students/31"
                rel="noopener noreferrer"
                style={
                  {
                    "fontSize": 14,
                    "fontWeight": "bold",
                  }
                }
                target="_blank"
              >
                Winnie
                 
                Kenobi
              </a>
            </div>
          </td>
          <td>
            <span
              style={
                {
                  "textAlign": "center",
                }
              }
            >
              Yes
            </span>
          </td>
          <td>
            <span
              style={
                {
                  "display": "inline-block",
                  "padding": 8,
                }
              }
            >
              0
            </span>
          </td>
          <td>
            <span
              style={
                {
                  "display": "inline-block",
                  "padding": 8,
                }
              }
            >
              0
            </span>
          </td>
          <td />
          <td />
          <td />
          <td />
          <td />
          <td />
          <td />
          <td />
          <td />
          <td />
          <td />
          <td />
        </tr>
        <tr>
          <td>
            <div
              style={
                {
                  "alignItems": "center",
                  "display": "flex",
                  "justifyContent": "flex-start",
                }
              }
            >
              <div
                className="StudentPhotoCropped"
                style={
                  {
                    "backgroundColor": "#ddd",
                    "backgroundImage": "url(/students/33/photo)",
                    "backgroundPosition": "center top",
                    "backgroundPositionY": -10,
                    "backgroundRepeat": "no-repeat",
                    "backgroundSize": "150%",
                    "border": "1px solid #ddd",
                    "borderRadius": 3,
                    "height": 58,
                    "marginRight": 10,
                    "overflow": "hidden",
                    "width": 58,
                  }
                }
              />
              <a
                href="/students/33"
                rel="noopener noreferrer"
                style={
                  {
                    "fontSize": 14,
                    "fontWeight": "bold",
                  }
                }
                target="_blank"
              >
                Winnie
                 
                Poppins
              </a>
            </div>
          </td>
          <td>
            <span
              style={
                {
                  "textAlign": "center",
                }
              }
            >
              No
            </span>
          </td>
          <td>
            <span
              style={
                {
                  "display": "inline-block",
                  "padding": 8,
                }
              }
            >
              0
            </span>
          </td>
          <td>
            <span
              style={
                {
                  "display": "inline-block",
                  "padding": 8,
                }
              }
            >
              0
            </span>
          </td>
          <td />
          <td />
          <td />
          <td />
          <td />
          <td />
          <td />
          <td />
          <td />
          <td />
          <td />
          <td />
        </tr>
        <tr>
          <td>
            <div
              style={
                {
                  "alignItems": "center",
                  "display": "flex",
                  "justifyContent": "flex-start",
                }
              }
            >
              <div
                className="StudentPhotoCropped"
                style={
                  {
                    "backgroundColor": "#ddd",
                    "backgroundImage": "url(/students/34/photo)",
                    "backgroundPosition": "center top",
                    "backgroundPositionY": -10,
                    "backgroundRepeat": "no-repeat",
                    "backgroundSize": "150%",
                    "border": "1px solid #ddd",
                    "borderRadius": 3,
                    "height": 58,
                    "marginRight": 10,
                    "overflow": "hidden",
                    "width": 58,
                  }
                }
              />
              <a
                href="/students/34"
                rel="noopener noreferrer"
                style={
                  {
                    "fontSize": 14,
                    "fontWeight": "bold",
                  }
                }
                target="_blank"
              >
                Olaf
                 
                White
              </a>
            </div>
          </td>
          <td>
            <span
              style={
                {
                  "textAlign": "center",
                }
              }
            >
              No
            </span>
          </td>
          <td>
            <span
              style={
                {
                  "display": "inline-block",
                  "padding": 8,
                }
              }
            >
              0
            </span>
          </td>
          <td>
            <span
              style={
                {
                  "display": "inline-block",
                  "padding": 8,
                }
              }
            >
              0
            </span>
          </td>
          <td />
          <td />
          <td />
          <td>
            <span
              style={
                {
                  "backgroundColor": "rgb(255, 231, 214)",
                  "color": "#666",
                  "cursor": "default",
                  "display": "inline-block",
                  "padding": 8,
                  "textAlign": "center",
                  "width": 35,
                }
              }
              title="PHYSICS 1"
            >
              D
            </span>
          </td>
          <td />
          <td />
          <td />
          <td />
          <td />
          <td />
          <td />
          <td>
            <span
              style={
                {
                  "backgroundColor": "rgba(51, 51, 51, 0.5)",
                  "color": "white",
                  "display": "inline-block",
                  "fontSize": 12,
                  "height": "95%",
                  "padding": 4,
                  "textAlign": "center",
                  "width": "95%",
                }
              }
            >
              2Way Spanish
            </span>
          </td>
        </tr>
        <tr>
          <td>
            <div
              style={
                {
                  "alignItems": "center",
                  "display": "flex",
                  "justifyContent": "flex-start",
                }
              }
            >
              <div
                className="StudentPhotoCropped"
                style={
                  {
                    "backgroundColor": "#ddd",
                    "backgroundImage": "url(/students/41/photo)",
                    "backgroundPosition": "center top",
                    "backgroundPositionY": -10,
                    "backgroundRepeat": "no-repeat",
                    "backgroundSize": "150%",
                    "border": "1px solid #ddd",
                    "borderRadius": 3,
                    "height": 58,
                    "marginRight": 10,
                    "overflow": "hidden",
                    "width": 58,
                  }
                }
              />
              <a
                href="/students/41"
                rel="noopener noreferrer"
                style={
                  {
                    "fontSize": 14,
                    "fontWeight": "bold",
                  }
                }
                target="_blank"
              >
                Mowgli
                 
                Kenobi
              </a>
            </div>
          </td>
          <td>
            <span
              style={
                {
                  "textAlign": "center",
                }
              }
            >
              Yes
            </span>
          </td>
          <td>
            <span
              style={
                {
                  "display": "inline-block",
                  "padding": 8,
                }
              }
            >
              0
            </span>
          </td>
          <td>
            <span
              style={
                {
                  "display": "inline-block",
                  "padding": 8,
                }
              }
            >
              0
            </span>
          </td>
          <td />
          <td />
          <td />
          <td />
          <td />
          <td />
          <td />
          <td />
          <td />
          <td />
          <td />
          <td>
            <span
              style={
                {
                  "backgroundColor": "rgba(51, 51, 51, 0.5)",
                  "color": "white",
                  "display": "inline-block",
                  "fontSize": 12,
                  "height": "95%",
                  "padding": 4,
                  "textAlign": "center",
                  "width": "95%",
                }
              }
            >
              2Way Spanish
            </span>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
